packages/nimble-components/src/table/specs/table-column-specs/table-column-menu-button.md

```diff
@@ -0,0 +1,205 @@
+
+The `menu-button-beforetoggle` and `menu-button-toggle` event detail will both include the original event fired by the menu button. With this, the client can interact with the original event if necessary, such as cancelling the event once the menu button supports this (see [nimble issue #1157](https://github.com/ni/nimble/issues/1157)).
+
+### Focus Recycling
+
+Because the menu button in the cell can have focus, we must override `focusedRecycleCallback()` in our cell view and have it call `blur()` on the button. If we don't, the focus can pass to other cells as you scroll.
+
+Note, there are open issues associated with scrolling, keyboard navigation, and focus recycling with the menu button.
+
+### Sizing
+
+The column will support fractional width sizing along with a minimum pixel width.
+
+### Interactions
+
+When the menu button is focused, it will have the same keyboard interactions as the `nimble-menu-button` outside of a table. Interacting with the menu button and its slotted menu, whether via keyboard or mouse, will not modify the selection in the table.
+
+### Accessibility
+
+If the text of the button does not fit fully in the button, it will be ellipsized and the button's `title` will be set to the full text.
+
+All other accessibility needs are already implemented by the table, menu button, and menu.
+
+### Angular integration
+
+An Angular wrapper will be created for the component. There are no special considerations for Angular.
+
+### Blazor integration
+
+A Blazor wrapper will be created for the component. There are no special considerations needed for Blazor.
+
+### API Updates to Existing Table Components
+
+Multiple API changes need to be made to the table, its subcomponents, and the classes it uses to facilitate slotting a menu through the table and into a specific cell view. This includes the following:
+
+-   The `ColumnInternalsOptions` interface will be updated to include an optional string array named `slotNames`, which allows a column to specify the names of any slots that need to be forwarded into a cell.
+-   As with the `ColumnInternalsOptions`, the `ColumnInternals` class will be updated to have a `slotNames` array that specifies the names of any slots that need to be forwarded into a cell. The value will be readonly and will be populated by the `ColumnInternalsOptions` `slotNames` property. It will default to an empty array if a value is not provided in `ColumnInternalsOptions`.
+-   Add slots to component templates:
+    -   table template - slots created within each table row element that has most recently requested the slot
+        -   slot's name: specified by client's column configuration (e.g. `software-version-menu`)
+        -   slot's slot: _unique-column-id_ + _column-requested-slot-name from column internal's slotNames array_ (e.g. table-column-slot2-menu)
```

Comment:
`_unique-column-id_ + _column-requested-slot-name` this is being done so that multiple columns can share the same `nimble-menu`? If that's the case it makes sense, think that intention should be captured somewhere, maybe the example usage?

Reply:
> _unique-column-id_ + _column-requested-slot-name this is being done so that multiple columns can share the same nimble-menu?

Yes. I wanted the code to be robust enough to handle having the same slot be specified on multiple columns, partly to be flexible for clients and partly to avoid having to validate that slot names were unique across all columns.

> If that's the case it makes sense, think that intention should be captured somewhere, maybe the example usage?

Agreed. I'll update the HLD to me more explicit about this.